src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -0,0 +1,278 @@
+        Instant executionStartTime,
+        String errorMessage,
+        String taskState,
+        AnomalyDetector detector
+    ) {
+        String detectorId = detector.getDetectorId();
+        try {
+            IntervalTimeConfiguration windowDelay = (IntervalTimeConfiguration) detector.getWindowDelay();
+            Instant dataStartTime = detectionStartTime.minus(windowDelay.getInterval(), windowDelay.getUnit());
+            Instant dataEndTime = executionStartTime.minus(windowDelay.getInterval(), windowDelay.getUnit());
+            User user = detector.getUser();
+
+            AnomalyResult anomalyResult = new AnomalyResult(
+                detectorId,
+                null, // no task id
+                new ArrayList<FeatureData>(),
+                dataStartTime,
+                dataEndTime,
+                executionStartTime,
+                Instant.now(),
+                errorMessage,
+                null, // single-stream detectors have no entity
+                user,
+                anomalyDetectionIndices.getSchemaVersion(ADIndex.RESULT),
+                null // no model id
+            );
+            String resultIndex = detector.getResultIndex();
+            if (resultIndex != null && !anomalyDetectionIndices.doesIndexExist(resultIndex)) {
+                // Set result index as null, will write exception to default result index.
+                anomalyResultHandler.index(anomalyResult, detectorId, null);
+            } else {
+                anomalyResultHandler.index(anomalyResult, detectorId, resultIndex);
+            }
+
+            if (errorMessage.contains(CommonErrorMessages.NO_CHECKPOINT_ERR_MSG) && !detector.isMultiCategoryDetector()) {
+                // single stream detector raises ResourceNotFoundException containing CommonErrorMessages.NO_CHECKPOINT_ERR_MSG
+                // when there is no checkpoint.
+                // Delay real time cache update by one minute so we will have trained models by then and update the state
+                // document accordingly.
+                threadPool.schedule(() -> {
+                    RCFPollingRequest request = new RCFPollingRequest(detectorId);
+                    client.execute(RCFPollingAction.INSTANCE, request, ActionListener.wrap(rcfPollResponse -> {
+                        long totalUpdates = rcfPollResponse.getTotalUpdates();
+                        // if there are updates, don't record failures
+                        updateLatestRealtimeTask(
+                            detectorId,
+                            taskState,
+                            totalUpdates,
+                            detector.getDetectorIntervalInMinutes(),
+                            totalUpdates > 0 ? "" : errorMessage
+                        );
+                    }, e -> {
+                        log.error("Fail to eecute RCFRollingAction", e);
+                        updateLatestRealtimeTask(detectorId, taskState, null, null, errorMessage);
```

Comment:
should the realtime task error message be `e` instead of `errorMessage`? The latter will always be the one containing `CommonErrorMessages.NO_CHECKPOINT_ERR_MSG`. I'm not sure if it's more appropriate to keep that as the error message here, or if it's more useful to use the `e` error as returned by the polling failure. Maybe the `e` could be too unrelated to issues with the detector itself since it could be an issue with cluster, a particular node, etc.

Reply:
yeah, e could be too unrelated to issues with the detector itself since it could be an issue with cluster, a particular node, etc. errorMessage is the one due to running AnomalyResultAction and the real time cache mostly records that.